tests/data_context/test_data_context_utils.py

```diff
@@ -12,3 +13,217 @@ def test_load_class_raises_error_when_module_not_found():
 def test_load_class_raises_error_when_class_not_found():
     with pytest.raises(gee.PluginClassNotFoundError):
         load_class("TotallyNotARealClass", "great_expectations.datasource")
+
+
+def test_password_masker_mask_db_url():
+    """
+    What does this test and why?
+    The PasswordMasker.mask_db_url() should mask passwords consistently in database urls.
+    This test uses database url examples from
+    https://docs.sqlalchemy.org/en/14/core/engines.html#database-urls
+    """
```

Comment:
Thank you so much for the thorough tests.  One subtlety that wasn't immediately obvious for me was why the output for `use_urlparse=True` and `use_urlparse=False` (default) were the same each time.  Of course they would have to be... whether you go through SQLAlchemy or urlparse you would want the same results.  

One very small ask is : we add an additional sentence under `What does this test and why?` that describes this?